### PR TITLE
Allow universal analytics (analytics.js) to be used without WooCommerce

### LINKED ISSUE #9129: Google Analytics Tracking Code Placed Incorrectly
#### Summary

User reports that this issue causes trouble with Google Analytics tracking. 

Referenced in 1036031-zen.

#### Steps to reproduce the issue

0. Starting from a test site with Google Analytics tracking: https://joentest1.mystagingwebsite.com/
1. View page source
2. Search for `<!-- Jetpack Google Analytics -->`

#### What I expected

Google Analytics tracking code is shown as outlined in [this article](https://support.google.com/analytics/answer/1008080?hl=en&visit_id=1-636573456400247366-2318498962&rd=1):

> right after the `<head>` tag on each page of your site. 

#### What happened instead

Google Analytics code is not inside `<head>` and instead appears on line 328 (on the test site).

screen shot 2018-03-22 at 21 39 07

### PR DESCRIPTION
**Pull request by allendav**
Fixes #9129

#### Changes proposed in this Pull Request:

* Adds a filter to allow end users to force-activate Universal Analytics (analytics.js) even if WooCommerce is not present, e.g.

`add_filter( 'jetpack_wga_use_universal_analytics', '__return_true' );`

* Initializes singleton in init action to give time for plugins/themes to use that filter

#### Testing instructions:

* Install and connect Jetpack using this branch
* Install WooCommerce
* Enable Google Analytics Enhanced eCommerce Analytics in Jetpack Settings
* View source to ensure that analytics.js is enqueued in the head tag (at/near the top) (including the ec module request) and your eCommerce events (like Add to Cart) appear in your Javascript console Network tab (turn log preservation on)
* Deactivate WooCommerce
* Verify ga.js is enqueued instead (this is the "legacy" analytics JS)
* Add the filter above to force universal analytics on
* View source to ensure that analytics.js is enqueued in the head tag (at/near the top), including the pageview universal command (but not including the ec module command)

cc @joendotcom @georgestephanis @rtrevellyan 

IMPORTANT NOTE: It is the responsibility of the site owner to ensure their site's privacy policy correctly reflects the use of Google Analytics Universal Analytics when this filter is enabled.

### CODE REVIEW

**Review by oskosk — APPROVED** [review]
LGTM!

**Comment by oskosk** [comment]
@allendav what's the status of this one? Should it be release in 6.2 ?

**Comment by allendav** [comment]
Sorry @oskosk - lemme refresh this

**Comment by brbrr** [comment]
> Deactivate WooCommerce
> Verify ga.js is enqueued instead (this is the "legacy" analytics JS)
I wasn't able to reproduce this behavior. After disabling Woo - it still enqueues new `analytics.js` file. So applying that filter didn't make any effect


Although, I was able to enqueue `analytics.js` with provided filter on newly created site _without_ Woo activation.
Steps I followed:
- Create new JN site, connect Premium Jetpack
- enable Google analytics via calypso settings
- Check `<head>` tag on site frontend. make sure legacy `ga` is loaded.
- Add provided filter somewhere (I added it into `jetpack.php` file)
- Check `<head>` tag on site frontend. make sure that new `analytics.js` loaded.

File: modules/google-analytics/wp-google-analytics.php [hunk 1]
```diff
@@ -50,7 +50,18 @@ class Jetpack_Google_Analytics {
...
-		if ( Jetpack_Google_Analytics_Options::enhanced_ecommerce_tracking_is_enabled() ) {
+		/**
+		 * Allow for universal analytics (analytics.js) to be used outside of ecommerce contexts
+		 *
+		 * @since 6.5.0
+		 *
+		 * @param bool Whether universal analytics (analytics.js) should be used (true) or legacy (ga.js) should be used (false)
+		 */
+		$use_universal_analytics = apply_filters( 'jetpack_wga_use_universal_analytics',
+			Jetpack_Google_Analytics_Options::enhanced_ecommerce_tracking_is_enabled()
```
Comment (oskosk) [inline modules/google-analytics/wp-google-analytics.php]: Seems like this filter is defaulting to `true` if enhanced ecommerce is enabled. so the provided filter in the testing instructions would not affect the truthiness we're getting after the step: `Enable Google Analytics Enhanced eCommerce Analytics in Jetpack Settings`.

Please @allendav  clarify. @brbrr comments about not seeing `ga.js` being logged come from this same fact.

**Review by oskosk — CHANGES_REQUESTED** [review]

**Comment by oskosk** [comment]
I pushed a commit updating the version of introduction to `6.5.0` for the filters

**Comment by orenwolf** [comment]
any chance we can get some reviews on this change? We're already using analytics.js on our pages, and it'd be nice to use the jetpack integration rather than having to disable it because it will only give us ga.js options currently.

**Comment by kraftbj** [comment]
Hi @orenwolf this is in progress, which includes review and conversation about the code change. You can follow along for updates. Currently, it has been initially reviewed and awaiting reply from @allendav  to clarify some things.

**Pull request closed without merging**